packages/env/lib/env.js

```diff
@@ -4,235 +4,192 @@
+				}
+			} )(),
+
+			...config.pluginSources.map( ( source ) =>
+				downloadSource( source, { onProgress: getProgressSetter( source.basename ) } )
+			),
+
+			...config.themeSources.map( ( source ) =>
+				downloadSource( source, { onProgress: getProgressSetter( source.basename ) } )
+			),
 		] );
 
-		// Remove dangling containers and finish.
-		await dockerCompose.rm( dockerComposeOptions );
-		spinner.text = `Started WordPress@${ ref }.`;
+		spinner.text = 'Starting WordPress.';
+
+		await dockerCompose.upMany( [ 'wordpress', 'tests-wordpress' ], {
+			config: config.dockerComposeConfigPath,
+		} );
+
+		try {
+			await checkDatabaseConnection( config );
+		} catch ( error ) {
+			// Wait 30 seconds for MySQL to accept connections.
+			await retry( () => checkDatabaseConnection( config ), {
+				times: 30,
+				delay: 1000,
+			} );
+
+			// It takes 3-4 seconds for MySQL to be ready after it starts accepting connections.
+			await sleep( 4000 );
+		}
+
+		// Retry WordPress installation in case MySQL *still* wasn't ready.
+		await Promise.all( [
+			retry( () => configureWordPress( 'development', config ), { times: 2 } ),
+			retry( () => configureWordPress( 'tests', config ), { times: 2 } ),
+		] );
+
+		spinner.text = 'WordPress started.';
 	},
 
-	async stop( { spinner = {} } ) {
+	/**
+	 * Stops the development server.
+	 *
+	 * @param {Object} options
+	 * @param {Object} options.spinner A CLI spinner which indicates progress.
+	 */
+	async stop( { spinner } ) {
 		spinner.text = 'Stopping WordPress.';
-		await dockerCompose.stop( dockerComposeOptions );
+
+		const configPath = path.resolve( '.wp-env.json' );
+		const { dockerComposeConfigPath } = await readConfig( configPath );
+
+		await dockerCompose.down( { config: dockerComposeConfigPath } );
+
 		spinner.text = 'Stopped WordPress.';
 	},
 
+	/**
+	 * Wipes the development server's database, the tests server's database, or both.
+	 *
+	 * @param {Object} options
+	 * @param {string} options.environment The environment to clean. Either 'development', 'tests', or 'all'.
+	 * @param {Object} options.spinner A CLI spinner which indicates progress.
+	 */
 	async clean( { environment, spinner } ) {
-		const context = await detectContext();
-		const dependencies = await resolveDependencies();
-		const activateDependencies = () =>
-			Promise.all( dependencies.map( activateContext ) );
-
-		const description = `${ environment } environment${
-			environment === 'all' ? 's' : ''
-		}`;
+		const description = `${ environment } environment${ environment === 'all' ? 's' : '' }`;
 		spinner.text = `Cleaning ${ description }.`;
 
-		// Parallelize task sequences for each environment.
+		const configPath = path.resolve( '.wp-env.json' );
+		const config = await readConfig( configPath );
+
+		await fs.mkdir( config.workDirectoryPath, { recursive: true } );
+
+		await fs.writeFile(
+			config.dockerComposeConfigPath,
+			yaml.dump( buildDockerComposeConfig( config ) )
+		);
```

Comment:
This seems common enough to build into a function. Especially if we decide to make new commands like `destroy`.